### PR TITLE
[BugFix] support Array<Array<>> of Parquet file

### PR DESCRIPTION
**Pull request by zombee0**
Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
For array type, the physical_column_index of field is 0, we need to iterate the children and collect their io ranges.
we need check the child's type, if it still is Array, we need iterate deeper.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by dirtysalt** [comment]
Could you add some test cases?

**Comment by zombee0** [comment]
> Could you add some test cases?

thank you, I will.

File: be/src/formats/parquet/group_reader.cpp [hunk 1]
```diff
@@ -214,42 +214,38 @@ vectorized::ChunkPtr GroupReader::_create_read_chunk(const std::vector<int>& col
...
-    int64_t offset = 0;
-    if (column.__isset.dictionary_page_offset) {
-        offset = column.dictionary_page_offset;
+    // 1. We collect column io ranges for each row group to make up the shared buffer, so we get column
+    // metadata (such as page offset and compressed_size) from _row_group_meta directly rather than file_metadata.
+    // 2. For map or struct columns, the physical_column_index indicates the real column index in rows group meta,
+    // and it may not be equal to col_idx_in_chunk.
+    // 3. For array type, the physical_column_index is 0, we need to iterate the children and
+    // collect their io ranges.
+    if (field.type.type == TYPE_ARRAY) {
```
Comment (GavinMar) [inline be/src/formats/parquet/group_reader.cpp:232]: You'd better add some ut for parquet format that the array elements are still arrays.

**Review by DorianZheng — COMMENTED** [review]
LGTM, plz add ut

**Review by GavinMar — APPROVED** [review]
LGTM

File: be/src/formats/parquet/group_reader.cpp [hunk 2]
```diff
@@ -214,42 +214,38 @@ vectorized::ChunkPtr GroupReader::_create_read_chunk(const std::vector<int>& col
...
+        for (auto& child : field.children) {
```
Comment (DorianZheng) [inline be/src/formats/parquet/group_reader.cpp:233]: const auto&

File: be/src/formats/parquet/group_reader.cpp [hunk 3]
```diff
@@ -214,42 +214,38 @@ vectorized::ChunkPtr GroupReader::_create_read_chunk(const std::vector<int>& col
...
+            _collect_field_io_range(child, ranges, end_offset);
+        }
     } else {
-        offset = column.data_page_offset;
+        auto& column = _row_group_metadata->columns[field.physical_column_index].meta_data;
```
Comment (DorianZheng) [inline be/src/formats/parquet/group_reader.cpp:237]: const auto&

**Review by dirtysalt — APPROVED** [review]
LGTM

**Review by DorianZheng — APPROVED** [review]
left comments, rest LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/formats/parquet/group_reader.cpp (hunk 2 above)
Comment (zombee0) [reply be/src/formats/parquet/group_reader.cpp:233]: got it, I will adopt it in the map_parquet pr.

**Pull request merged**